src/Exception/BadRequestParamHttpException.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Exception;
+
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+
+final class BadRequestParamHttpException extends BadRequestHttpException
+{
+    /**
+     * @param mixed $value
+     */
+    public function __construct(string $name, string $expectedType, $value)
```

Comment:
As we are using multiple expected types in some cases, I think the argument 2 should also accept an array in order to build the messages for the accepted types here. Otherwise, there is no way to ensure all the exception messages are respecting the same syntax (`srting|int`, `[string|int]`, `(string|int)`, `string, int`, `string and int`, etc).

Reply:
I did something similar to the UnexpectedValueException from symfony https://github.com/symfony/validator/blob/5.4/Exception/UnexpectedValueException.php#L21

but I can add array if you prefer.

Reply:
Yes, I know. But in that case, the argument is expected to be single. Here we already know that multiple types can be received.